packages/next/src/export/worker.ts

```diff
@@ -345,21 +320,48 @@ async function exportPageImpl(input: ExportPageInput) {
 
 export default async function exportPage(
   input: ExportPageInput
-): Promise<ExportPageResults> {
+): Promise<ExportPageResult | undefined> {
   // Configure the http agent.
   setHttpClientAndAgentOptions({
     httpAgentOptions: input.httpAgentOptions,
   })
 
+  const files: ExportedPageFile[] = []
+  const baseFileWriter: FileWriter = async (
+    type,
+    path,
+    content,
+    encodingOptions = 'utf-8'
+  ) => {
+    await fs.mkdir(dirname(path), { recursive: true })
+    await fs.writeFile(path, content, encodingOptions)
+    files.push({ type, path })
+  }
+
```

Comment:
Ideally with this structure, we can leverage the `files` returned here as the "output" of the export, and avoid having to _guess_ the files exported by doing string manipulation.